.github/workflows/e2e-tests.yml

```diff
@@ -124,7 +124,7 @@ jobs:
         pnpm run test:run "${TEST_ARGS[@]}"
 
     - name: Send Slack notification
-      if: ${{ ! cancelled() }}
+      if: ${{ failure() && ! cancelled() }}
```

Comment:
If we can do this for our old Slack notification, couldn't we do the same for the new one, without having to return early from within the action?

Reply:
That would be easier indeed. But my plan there is to also run the action on success to notify if the tests passed on a rerun.

Reply:
That makes sense. It may be good to document that somewhere in the file maybe, so we can quickly understand the differences between the different steps in a few months?

Reply:
Good idea. Added some comments with [53dff83](https://github.com/Automattic/jetpack/pull/25437/commits/53dff8345b098b64da50f84897c0a5e7a6fad392)